pom.xml

```diff
@@ -302,19 +302,13 @@
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-jar-plugin</artifactId>
                 <version>${maven.jar.version}</version>
-                <!--
                 <configuration>
                     <archive>
-                        <manifest>
```

Comment:
Why were these two entries removed?

Reply:
I was not sure about those entries .. I removed them to not introduce more changes than necessary to the current state. As they have been there before they might be safe.

Reply:
I'm confused 😄 

How is OSGi supported per this PR?  I don't see the bundle plugin being used, so I'm not exactly sure what is going on here 😅 

Also, can you please re-enable these lines and see if things still work?  I'd like to retain that metadata if it doesn't hurt anything.

Reply:
I reenabled the two other settings now.
The OSGi headers are created in two steps:
1. The maven-bundle-plugin creates an enhanced Manifest file at https://github.com/jwtk/jjwt/blob/c246385be1b4beca4a5cd1a2debc1f5aa5008848/pom.xml#L432-L457
2. The jar plugin then must be configured to use this changed Manifest

The current state on master is that only step 1 happens.
So my PR is meant to reenable step 2 again.

You can check the effect by looking into the Manifest of the created jar files before and after the change.

Reply:
Ah, I forgot about the bundle plugin in the pom - thanks for clarifying.

That said, will this configuration work for every module now that we've gone to a multi-module project structure?

That is, we have an `api` module, an `impl` module, and a module for each of the JSON processor implementations (`jackson`, `orgjson`, `gson`).  Is the OSGi metadata valid for all of them?  Or does the bundle plugin configuration need to be re-defined for each module?

Reply:
Normally the default configuration works fine. I will give feedback if we spot issues.

Reply:
I meant the current configuration of the bundle plugin:

```
                <configuration>
                    <instructions>
                        <Import-Package><![CDATA[
                            android.util;resolution:=optional,
                            org.bouncycastle.jce;resolution:=optional,
                            org.bouncycastle.jce.spec;resolution:=optional,
                            *
                        ]]>
                        </Import-Package>
                    </instructions>
                </configuration>
```

Is this valid for every module in the project including the new module-per-json-processor approach?

Reply:
Ah .. got it. 
I think I can achieve the same without the config. 
If I make the dependencies optional in maven then the import will also be optional.
I will create another commit for this.

Reply:
to generate the OSGi metadata, the maven bundle plugin should be present in each pom
1. api
2. impl
3. each of json processor pom

not sure if Christian has some other way e.g. bnd maven plugin

Reply:
The settings in the parent are applied to all sub modules too. 
I checked the packages created and the optional imports above do not seem to be needed. So I removed this config.
As there might be the need to tweak individual modules later I added a config to support an optional bnd.bnd file per module. For now it is not needed but it makes sure the parent does not have to be changed later.